### PR TITLE
FINERACT-853 Use find-sec-bugs SpotBugs plugin to detect SQL injectio…

### PR DESCRIPTION
**Pull request by nnatarajan**
…n issues (and other security related problems)

## Description
Describe the changes made and why they were made. Ignore if these details are present on the associated Jira ticket

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [ ] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [ ] API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm has been updated with details of any API changes.

- [ ] Integration tests have been created/updated for verifying the changes made.

- [ ] All Integrations tests are passing with the new commits.

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

**Comment by thesmallstar** [comment]
@nnatarajan  you have spotless violations in your code in the files you didn't change. I am not sure what's going on, but maybe you can run ./gradlew spotlessApply and rebase this?

**Comment by thesmallstar** [comment]
This should not fail due to "Spotless" once #1091 is merged (and this branch is rebased).

**Comment by nnatarajan** [comment]
@thesmallstar all the checks ran ok before I sent PR, I will look into it

**Comment by nnatarajan** [comment]
@thesmallstar , thank you for letting me know.

**Comment by nnatarajan** [comment]
@thesmallstar, I appreciate your help. I ran ./gradlew spotlessApply, which did update five files. I will wait for 1091 merge.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareTenantIdentifierFilter.java
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareTenantIdentifierFilter.java:100]: I don't think this is correct? But it's been YEARS since I've last dealt with CORS...

**Review by vorburger — CHANGES_REQUESTED** [review]
see prior discussion on #749

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareTenantIdentifierFilter.java
Comment (nnatarajan) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareTenantIdentifierFilter.java:100]: @vorburger  When I ran this check locally, the build passed all tests. Can you let me know what you are looking for specifically?

Comment (nnatarajan) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareTenantIdentifierFilter.java:100]: The files updated were related PR 1091. After the fact,  the same updates were requested on the develop branch. The failing checks were independent of this PR.

**Comment by nnatarajan** [comment]
@vorburger , it seems that both checks passed with rebase. Please let me know if I should investigate CORS further. I don't mind if this is a concern.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareTenantIdentifierFilter.java
Comment (nnatarajan) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareTenantIdentifierFilter.java:100]: The star must be replaced by a string and I entered what seemed most logical in the class.

Comment (vorburger) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareTenantIdentifierFilter.java:100]: >  entered what seemed most logical in the class.

I don't think this hard-coded String (`Fineract-Platform-TenantId`) makes any sense at all, and I suspect that this would break UI clients.

**Review by vorburger — CHANGES_REQUESTED** [review]
NOK

**Comment by thesmallstar** [comment]
@nnatarajan do you know if some reports are generated for possible SQL injection codes? Do we need to add any config? If no, does this really help to find SQL injections? (since the build passes without any significant changes in the code).

**Comment by nnatarajan** [comment]
@thesmallstar , I just found out that this was actually your issue and I mistakenly took this on. I am going to close this PR as you are much more familiar with this.

**Comment by thesmallstar** [comment]
no no! you can pick this up, I did some research so I had those doubts. I thought maybe you could answer them. I barely know anything about this :P

**Pull request closed without merging**
